core/utils/dom.js

```diff
@@ -72,16 +72,132 @@ Blockly.utils.dom.cacheReference_ = 0;
  */
 Blockly.utils.dom.canvasContext_ = null;
 
+/**
+ * A name with the type of the SVG element stored in the generic.
+ * @param {string} tagName The SVG element tag name.
+ * @constructor
+ * @template T
+ */
+Blockly.utils.dom.SVGElementType = function(tagName) {
+  /**
+   * @type {string}
+   * @private
+   */
+  this.tagName_ = tagName;
+};
+
+/**
+ * Returns the SVG element tag name.
+ * @return {string} The name.
+ * @override
+ */
+Blockly.utils.dom.SVGElementType.prototype.toString = function() {
+  return this.tagName_;
+};
+
+/** @type {!Blockly.utils.dom.SVGElementType<!SVGAnimateElement>} */
+Blockly.utils.dom.SVGElementType.SVGAnimateElement =
+    new Blockly.utils.dom.SVGElementType('animate');
+
+/** @type {!Blockly.utils.dom.SVGElementType<!SVGCircleElement>} */
+Blockly.utils.dom.SVGElementType.SVGCircleElement =
+    new Blockly.utils.dom.SVGElementType('circle');
+
+/** @type {!Blockly.utils.dom.SVGElementType<!SVGClipPathElement>} */
+Blockly.utils.dom.SVGElementType.SVGClipPathElement =
+    new Blockly.utils.dom.SVGElementType('clipPath');
+
+/** @type {!Blockly.utils.dom.SVGElementType<!SVGDefsElement>} */
+Blockly.utils.dom.SVGElementType.SVGDefsElement =
+    new Blockly.utils.dom.SVGElementType('defs');
+
+/** @type {!Blockly.utils.dom.SVGElementType<!SVGFECompositeElement>} */
+Blockly.utils.dom.SVGElementType.SVGFECompositeElement =
```

Comment:
I think the camel casing for all of these should be
`SvgElementType` (keeping with the pre-existing function style like `createSvgElement` and https://google.github.io/styleguide/jsguide.html#naming-camel-case-defined)

For all of the ones that start with `fe` I think the rules might be a little ambiguous but personally I think `SvgFeCompositeElement` (or `FeComposite`) is a bit more readable but this one's more up to interpretation I think.

Reply:
Similarly to the previous comment, the value of the end of the type there is to use the name as an indicator to the exact type you're getting. Being able to infer that `Blockly.utils.dom.SVGElementType.SVGFECompositeElement` returns an element of type `SVGFECompositeElement`. 

I agree that `Blockly.utils.dom.SVGElementType` should change to `Blockly.utils.dom.SvgElementType`.

Reply:
Sorry, I think I confused myself, but here I was only nitpicking the capitalization of `Svg` and `Fe` in the interest of readability. But I think I didn't realize that these types already existed so you can't change their names, so this makes a lot more sense now :)

But while I am on the topic of nitpicking capitalization, if you are choosing constant case for the names now, then should there be underscores between words? e.g. `CLIP_PATH` instead of `CLIPPATH` ?

I find a lot of capital letters in a row hard to read esp. since I'm not super familiar with SVG types so the words don't stand out easily :)

Reply:
Normally I would agree with sticking with constant case, but I think in this case, there's an advantage of being able to directly convert from something like the tag name (eg: `clipPath`) to type name (eg: CLIPPATH), without having to wonder if there is an underscore.

Reply:
Personally, I think using a non-standard format of constant case actually makes it harder, because following the normal algorithm I would expect there to be an underscore at the camelCase boundaries. But I will defer to your judgment :)